### PR TITLE
[REVIEW] CSV Reader: Change the meaning of the quoting parameter to match Pandas

### LINKED ISSUE #974: [BUG] Option `quoting` in `read_csv` behaves different from pandas
In cudf read_csv() the option `quoting` is a Boolean that specifies whether quotation marks should be removed or kept in strings.  It has no impact on parsing itself.

In pandas the option `quoting` can take on one of four values as shown below.  Only `QUOTE_NONE` is relevant to read_csv().  If `quoting=QUOTE_NONE` quotation marks should be treated like regular characters.  This is the same behavior as for `quotechar=\'0'`.

```
csv.QUOTE_ALL (0)
Instructs writer objects to quote all fields.

No special behavior for reader.

csv.QUOTE_MINIMAL (1)
Instructs writer objects to only quote those fields which contain special characters such as delimiter, quotechar or any of the characters in lineterminator.

No special behavior for reader.

csv.QUOTE_NONNUMERIC (2)
Instructs writer objects to quote all non-numeric fields.

No special behavior for reader.

csv.QUOTE_NONE (3)
Instructs writer objects to never quote fields. When the current delimiter occurs in output data it is preceded by the current escapechar character. If escapechar is not set, the writer will raise Error if any characters that require escaping are encountered.

```
Instructs reader to perform no special processing of quote characters.

### PR DESCRIPTION
**Pull request by vuule**
Fixes #974

### CODE REVIEW

**Comment by vuule** [comment]
rerun tests

**Comment by mjsamoht** [comment]
Please change the `quoting` argument to be an enum like pandas.  Otherwise, LGTM.

**Comment by vuule** [comment]
@mjsamoht  Modified the API based on the review, please take another look

File: python/cudf/io/csv.py [hunk 1]
```diff
@@ -380,7 +393,7 @@ def read_csv(filepath_or_buffer, lineterminator='\n',
 
 
 def read_csv_strings(filepath_or_buffer, lineterminator='\n',
-                     quotechar='"', quoting=True, doublequote=True,
+                     quotechar='"', quoting='minimal', doublequote=True,
```
Comment (kkraus14) [inline python/cudf/io/csv.py]: Update default to 0 here as well
```suggestion
                     quotechar='"', quoting=0, doublequote=True,
```

File: python/cudf/tests/test_csvreader.py [hunk 2]
```diff
@@ -229,7 +229,7 @@ def test_csv_reader_strings_quotechars(tmpdir):
         fp.write('\n'.join(lines) + '\n')
 
     cols = read_csv_strings(str(fname), names=names, dtype=dtypes, skiprows=1,
-                            quotechar='\"', quoting=True)
+                            quotechar='\"', quoting='all')
```
Comment (kkraus14) [inline python/cudf/tests/test_csvreader.py]: Change to integer or csv.XXX parameter

File: python/cudf/io/csv.py [hunk 3]
```diff
@@ -28,8 +28,20 @@ def is_file_like(obj):
...
+_quoting_enum = {
+    'minimal': libgdf.QUOTE_MINIMAL,
+    0: libgdf.QUOTE_MINIMAL,
+    'all': libgdf.QUOTE_ALL,
+    1: libgdf.QUOTE_ALL,
+    'nonnumeric': libgdf.QUOTE_NONNUMERIC,
+    2: libgdf.QUOTE_NONNUMERIC,
+    'none': libgdf.QUOTE_NONE,
+    3: libgdf.QUOTE_NONE,
+}
```
[the remark this answers is not in the document]
Comment (kkraus14) [reply python/cudf/io/csv.py]: Confirmed it does not, the strings should be removed from here.

**Comment by vuule** [comment]
@kkraus14 Addressed all of your comments, please review.

**Pull request merged**